connector_magento/components/backend_adapter.py

```diff
@@ -153,9 +153,20 @@ def call(self, method, arguments, http_method=None, storeview=None):
             # record(method, arguments, result)
             return result
         except (socket.gaierror, socket.error, socket.timeout) as err:
-            raise NetworkRetryableError(
-                'A network error caused the failure of the job: '
-                '%s' % err)
+            if err.response is not None:
+                if err.response.headers.get('Content-Type') == 'application/json':
```

Comment:
There is an approach that is considered nicely Pythonic, which is "it's easier to ask for forgiveness than permission", or "EAFP". See https://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html

Because we don't care about the content type or anything. We just want to get the json if there is one, and do something else otherwise. So without checking the content type, you can simply do

```
try:
    json = err.response.json()
    <do something with the json>
except Exception:
    pass
<do someting else>
```

It is not nice to catch the generic `Exception`, so you should replace it with the exception that response.json() actually throws. That is documented here: https://docs.python-requests.org/en/latest/user/quickstart/#json-response-content